code/modules/power/singularity/field_generator.dm

```diff
@@ -196,44 +183,37 @@ field_generator power level display
 
 
 /obj/machinery/field/generator/proc/calc_power()
-	if(Varpower)
-		return 1
-
 	update_icon()
-	if(src.power > field_generator_max_power)
-		src.power = field_generator_max_power
+	if(power > field_generator_max_power)
+		power = field_generator_max_power
 
 	var/power_draw = 2
 	for (var/obj/machinery/field/containment/F in fields)

```

Comment:
if there are only this type of objects in fields, you could use an istypeless"for" loop.